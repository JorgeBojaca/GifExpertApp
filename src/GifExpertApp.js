import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories);
  const addCategory = (cat) => {
    setCategories([cat, ...categories]);
  };
  return (
    <>
      <header>
        <h2>GifExpertApp</h2>
        <AddCategory addCategory={addCategory} />
        <hr />
      </header>

      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};

export default GifExpertApp;
