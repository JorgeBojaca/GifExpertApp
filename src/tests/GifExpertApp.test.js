import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas para <GifExpertApp />', () => {
  test('El componente se muestra correctamente', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });
  test('El componente debe renderizar dos GifGrid', () => {
    const categories = ['Giphy', 'Hello World'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
