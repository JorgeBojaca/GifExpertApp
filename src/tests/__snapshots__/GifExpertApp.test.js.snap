// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para <GifExpertApp /> El componente debe renderizar dos GifGrid 1`] = `
<Fragment>
  <header>
    <h2>
      GifExpertApp
    </h2>
    <AddCategory
      addCategory={[Function]}
    />
    <hr />
  </header>
  <GifGrid
    category="Giphy"
    key="Giphy"
  />
  <GifGrid
    category="Hello World"
    key="Hello World"
  />
</Fragment>
`;

exports[`Pruebas para <GifExpertApp /> El componente se muestra correctamente 1`] = `
<Fragment>
  <header>
    <h2>
      GifExpertApp
    </h2>
    <AddCategory
      addCategory={[Function]}
    />
    <hr />
  </header>
</Fragment>
`;
